mpqp/qasm/mpqp_to_qasm.py

```diff
@@ -0,0 +1,218 @@
+from __future__ import annotations
+
+import logging
+import numpy as np
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from mpqp.core.instruction import Instruction
+    from mpqp.core.circuit import QCircuit
+
+from mpqp.core.instruction.barrier import Barrier
+from mpqp.core.instruction.measurement import BasisMeasure
+from mpqp.core.instruction.breakpoint import Breakpoint
+from mpqp.core.instruction.measurement import ExpectationMeasure
+from mpqp.core.instruction.gates.gate import SingleQubitGate
+from mpqp.core.instruction.gates import *
+from mpqp.qasm.open_qasm_2_and_3 import remove_user_gates
+
+
+def _handle_measurement(instruction: BasisMeasure) -> str:
+    if instruction.c_targets is None:
+        return "\n".join(
+            f"measure q[{target}] -> c[{i}];"
+            for i, target in enumerate(instruction.targets)
+        )
+    else:
+        return "\n".join(
+            f"measure q[{target}] -> c[{c_target}];"
+            for target, c_target in zip(instruction.targets, instruction.c_targets)
+        )
+
+
+def _handle_connection(instruction: Gate | Barrier) -> str:
+    control = ""
+    if isinstance(instruction, ControlledGate):
+        control = ",".join([f"q[{j}]" for j in instruction.controls]) + ","
+    return control + ",".join([f"q[{j}]" for j in instruction.targets])
+
+
+def _float_to_str(f: float) -> str:
+    if f.is_integer():
+        return str(int(f))
+    elif f % np.pi == 0:
+        return f"{int(f/np.pi)}*pi" if f != np.pi else "pi"
+    else:
+        return f"pi/{int(1 / f * np.pi)}" if (np.pi * (1 / f)).is_integer() else str(f)
+
+
+def _handle_rotation(gate: ParametrizedGate) -> str:
```

Comment:
can't this be done at the gate level ? something like `Gate.to_other_language`
would eliminate the need for a separate function here

Reply:
This appliues to many of the private functions here